pubsub.go

```diff
@@ -61,19 +70,44 @@ func KeyToTopic(key string) string {
 // The constructor interface is complicated by the need to bootstrap the pubsub topic.
 // This could be greatly simplified if the pubsub implementation handled bootstrap itself
 func NewPubsubValueStore(ctx context.Context, host host.Host, cr routing.ContentRouting, ps *pubsub.PubSub, validator record.Validator) *PubsubValueStore {
-	return &PubsubValueStore{
+	psValueStore := &PubsubValueStore{
 		ctx: ctx,
 
-		ds:   dssync.MutexWrap(ds.NewMapDatastore()),
-		host: host, // needed for pubsub bootstrap
-		cr:   cr,   // needed for pubsub bootstrap
-		ps:   ps,
+		ds:                      dssync.MutexWrap(ds.NewMapDatastore()),
+		ps:                      ps,
+		host:                    host,
+		rebroadcastInitialDelay: 100 * time.Millisecond,
+		rebroadcastInterval:     time.Second,
 
 		subs:     make(map[string]*pubsub.Subscription),
 		watching: make(map[string]*watchGroup),
 
 		Validator: validator,
 	}
+
+	host.SetStreamHandler(OnJoinProto, func(s net.Stream) {
```

Comment:
I'm thinking this could actually be done in half a trip instead of an RTT by sending `{Identifier: Name, Latest : Data}` and not even caring about a response.

Thoughts?

Reply:
Also discussed with a larger audience to confirm libp2p/go-libp2p#698